suite/spdz2k/src/test/java/dk/alexandra/fresco/suite/spdz2k/AbstractSpdz2kTest.java

```diff
@@ -0,0 +1,92 @@
+package dk.alexandra.fresco.suite.spdz2k;
+
+import dk.alexandra.fresco.framework.ProtocolEvaluator;
+import dk.alexandra.fresco.framework.TestThreadRunner;
+import dk.alexandra.fresco.framework.TestThreadRunner.TestThreadFactory;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
+import dk.alexandra.fresco.framework.configuration.TestConfiguration;
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.network.async.AsyncNetwork;
+import dk.alexandra.fresco.framework.sce.SecureComputationEngine;
+import dk.alexandra.fresco.framework.sce.SecureComputationEngineImpl;
+import dk.alexandra.fresco.framework.sce.evaluator.BatchEvaluationStrategy;
+import dk.alexandra.fresco.framework.sce.evaluator.BatchedProtocolEvaluator;
+import dk.alexandra.fresco.framework.sce.evaluator.EvaluationStrategy;
+import dk.alexandra.fresco.suite.ProtocolSuiteNumeric;
+import dk.alexandra.fresco.suite.spdz2k.resource.Spdz2kResourcePool;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public abstract class AbstractSpdz2kTest<MarlinResourcePoolT extends Spdz2kResourcePool<?>> {
+
+  private final List<Integer> partyNumbers = Arrays.asList(2, 3);
+
+  void runTest(
+      TestThreadRunner.TestThreadFactory<MarlinResourcePoolT, ProtocolBuilderNumeric> f,
+      EvaluationStrategy evalStrategy) {
+    for (Integer numberOfParties : partyNumbers) {
+      runTest(f, evalStrategy, numberOfParties);
+    }
+  }
+
+  protected void runTest(
+      TestThreadFactory<MarlinResourcePoolT, ProtocolBuilderNumeric> f,
+      EvaluationStrategy evalStrategy, int noOfParties) {
+
+    List<Integer> ports = getFreePorts(2 * noOfParties);
```

Comment:
Is there any reason you cant just use the same network for both of these protocols?